### PR TITLE
476 - MGC docs re-org

### PR DESCRIPTION
**Pull request by jasonmadigan**
Ref: https://github.com/Kuadrant/multicluster-gateway-controller/issues/476

needs to land after: https://github.com/Kuadrant/multicluster-gateway-controller/issues/476

### CODE REVIEW

**Comment by jasonmadigan** [comment]
Screenshot 2023-09-06 at 13 19 16

File: mkdocs.yml [hunk 1]
```diff
@@ -132,19 +132,29 @@ nav:
       - limitador-operator/doc/logging.md
   - 'Multicluster Gateway Controller':
     - 'Overview': multicluster-gateway-controller/README.md
+    - 'Architecture': multicluster-gateway-controller/docs/architectural-overview.md
```
Comment (maleck13) [inline mkdocs.yml:135]: this doc is in the architecture repo. Is it possible to use that from there?

Comment (jasonmadigan) [reply mkdocs.yml:135]: Ah! Wondered where it was (that makes sense). Yep, I should be able to pull that in as well...

**Comment by jasonmadigan** [comment]
@openshift-ci-robot @openshift-merge-robot

**Comment by jasonmadigan** [comment]
Fixing some broken links.

**Pull request merged**